incubator/hnc/pkg/reconcilers/hierarchical_namespace.go

```diff
@@ -59,51 +59,121 @@ func (r *HierarchicalNamespaceReconciler) Reconcile(req ctrl.Request) (ctrl.Resu
 	nm := req.Name
 	pnm := req.Namespace
 
-	// Get instance from apiserver. If the instance doesn't exist, we don't want to reconcile
-	// it since it may trigger the HC reconciler to recreate the namespace that was just deleted.
-	// TODO expand on this to check the owner's hc.spec.allowCascadingDelete. If it's set to
-	//  true, we still want to reconcile the hns instance. See issue:
-	//  https://github.com/kubernetes-sigs/multi-tenancy/issues/501
+	// Get instance from apiserver. If the instance doesn't exist, do nothing and
+	// early exist because HCR watches HNS instances and (if applicable) has
+	// already updated the owner HC when this HNS was purged.
```

Comment:
What I meant in the comment is that when this HNS is purged, HCR will be triggered to update the HC instance (possilbly remove the finalizers if there's no HNSes). Since this is when the instance `IsNotFound()`, which means it's already purged, that's why I put that comment. WDYT?

Reply:
sg!